core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -166,20 +189,21 @@ public void delete(F deletedFile) {
   /**
    * Add a delete entry for a file.
    *
-   * <p>The entry's snapshot ID will be this manifest's snapshot ID. However, the original data
-   * sequence number of the file must be preserved when the file is marked as deleted.
+   * <p>The entry's snapshot ID will be this manifest's snapshot ID. However, the original data and
+   * file sequence numbers of the file must be preserved when the file is marked as deleted.
    *
    * @param deletedFile a file
    * @param dataSequenceNumber a data sequence number of the file (assigned when the file was added)
+   * @param fileSequenceNumber a file sequence number (assigned when the file was added)
    */
-  public void delete(F deletedFile, long dataSequenceNumber) {
-    addEntry(reused.wrapDelete(snapshotId, dataSequenceNumber, deletedFile));
+  public void delete(F deletedFile, long dataSequenceNumber, Long fileSequenceNumber) {
```

Comment:
It doesn't look like any of these delete methods are used, besides `delete(ManifestEntry)`. Do we need to introduce this new method or can we update the deprecation for `delete(ContentFile)` to point people to `delete(ManifestEntry)`?

Reply:
The problem is that `delete(ManifestEntry)` is package-private just like `ManifestEntry` itself. This, on the other hand, is a public method.